aspnetcore/fundamentals/http-requests.md

```diff
@@ -0,0 +1,392 @@
+
+Multiple handlers can be registered in the order that they should execute. Each handler wraps the next handler until the final `HttpClientHandler` executes the request.
+
+```csharp
+public static void Configure(IServiceCollection services)
+{
+    services.AddTransient<OuterHandler>();
+    services.AddTransient<InnerHandler>();
+
+    services.AddHttpClient("example", c =>
+    {
+        c.BaseAddress = new Uri("https://localhost:5000/");
+    })
+    .AddHttpMessageHandler<OuterHandler>() // This handler is on the outside and executes first on the way out and last on the way in.
+    .AddHttpMessageHandler<InnerHandler>(); // This handler is on the inside, closest to the request.
+}
+```
+
+## Handling errors with Polly
+
+_NOTE: This is a WIP since the Polly Extensions API is still currently under design._
+
+The preceding example demonstrated building a simple retry handler manually. A more robust and feature-rich approach is to leverage a popular third-party library called [Polly](https://github.com/App-vNext/Polly).
+
+Polly is a comprehensive resilience and transient-fault-handling library for .NET which allows developers to express policies such as Retry, Circuit Breaker, Timeout, Bulkhead Isolation, and Fallback in a fluent and thread-safe manner.
+
+Extension methods are provided for `HttpClientFactory` which enable easy integration and use of Polly policies with configured `HttpClient` instances. Rather than defining a handler manually, extension methods can be used which accept a Polly policy. This policy is used when executing HTTP requests.
+
+```csharp
+public static void Configure(IServiceCollection services)
+{
+    services.AddTransient<OuterHandler>();
+    services.AddTransient<InnerHandler>();
+
+    services.AddHttpClient("example", c =>
+    {
+        c.BaseAddress = new Uri("https://localhost:5000/");
+    })
+
+    // Build a totally custom policy using any criteria
+    .AddPolicyHandler(Policy.Handle<HttpRequestException>().RetryAsync())
+
+    // Build a policy that handles exceptions (connection failures)
+    .AddExceptionPolicyHandler(p => p.RetryAsync())
+
+    // Build a policy that handles exceptions and 500s from the remote server
+    .AddServerErrorPolicyHandler(p => p.RetryAsync())
+
+    // Build a policy that handles exceptions, 400s, and 500s from the remote server
+    .AddBadRequestPolicyHandler(p => p.RetryAsync());
+}
+```
+
+## DNS and handler rotation
+
+Each time `CreateClient` is called on the `HttpClientFactory` a new instance of a `HttpClient` is returned. The `HttpClientFactory` may reuse the underlying `HttpMessageHandler` when appropriate. The `HttpMessageHandler` is responsible for creating and maintainging the underlying operating system connection. Reusing the `HttpMessageHandler` avoids creating many connections on the host machine which can lead to socket exhaustion.
```

Comment:
- Add a comma before "a new instance of a"
- Add a comma after "on the host machine"
- avoids creating many --> avoids creating too many